src/webview/MessageList.js

```diff
@@ -180,13 +198,78 @@ class MessageList extends Component<Props> {
       showMessagePlaceholders,
     });
 
+    /**
+     * Effective URL of the MessageList webview.
+     *
+     * We provide all HTML at creation (or refresh) time; the document named
+     * here is not loaded, and doesn't even need to exist. It serves only as a
+     * placeholder, so that relative URLs and cross-domain security restrictions
+     * have somewhere to believe that this document originates from.
+     */
+    const baseUrl = `${assetsPath}/index.html`;
+
+    /**
+     * Paranoia^WSecurity: only load `baseUrl`, and only load it once. Any other
+     * requests should be handed off to the OS, not loaded inside the WebView.
+     */
+    const onShouldStartLoadWithRequest: (event: WebViewNavigation) => boolean = (() => {
+      // Inner closure to actually test the URL.
+      const urlTester: (url: string) => boolean = (() => {
+        // On Android this function is documented to be skipped on first load:
+        // therefore, simply never return true.
+        if (Platform.OS === 'android') {
+          return (url: string) => false;
+        }
+
+        // Otherwise (for iOS), return a closure that evaluates to `true` _exactly
+        // once_, and even then only if the URL looks like what we're expecting.
+        let loaded_once = false;
+        // The baseUrl, with its relative portion (if any) stripped.
+        const baseUrlTail = baseUrl.replace(/^\.\//, '');
+        // Disallow such monstrosities as `evilsite.com/?./webview/index.html`.
+        const unsafeUrlRegex = /[&?]/;
+        return (url: string) => {
+          if (!loaded_once) {
+            // The exact URL that will be loaded could be determined statically on
+            // Android. On iOS, though, it involves some unpredictable UUIDs which
+            // RN provides no good interface to. (`react-native-fs` is awful.)
+            if (
+              url.startsWith('file://')
+              && url.endsWith(baseUrlTail)
+              && !unsafeUrlRegex.test(url)
+            ) {
+              loaded_once = true;
+              return true;
+            }
+          }
+          return false;
+        };
+      })();
+
+      // Outer closure to perform logging.
+      return (event: WebViewNavigation) => {
+        const ok = urlTester(event.url);
+        if (!ok) {
+          logging.warn(`webview: rejected navigation event: ${JSON.stringify({ ...event })};
+    expected base URL = ${baseUrl}`);
+        }
+        return ok;
+      };
+    })();
+
+    // Note: neither `originWhitelist` nor `onShouldStartLoadWithRequest` are
+    // significant security barriers; they're checked only against the URL of
+    // the document itself. They cannot be used to validate the URL of any
+    // resource the WebView loads.
+    //
+    // Worse, as of at least v7.4.1, `originWhitelist` is completely broken: see
+    // https://github.com/react-native-community/react-native-webview/pull/697.
```

Comment:
"v7.4.1 of react-native-webview" would be helpful for interpreting the version number; even though it is possible to guess (as I'm guessing) based on that PR link :slightly_smiling_face:

Reply:
Removed the version number, instead. (Given the PR link, it's probably of very little utility.)